Yosemite/YosemiteTests/Stores/SitePluginStoreTests.swift

```diff
@@ -0,0 +1,78 @@
+import XCTest
+import Fakes
+
+@testable import Yosemite
+@testable import Networking
+@testable import Storage
+
+class SitePluginStoreTests: XCTestCase {
+    /// Mock Dispatcher
+    ///
+    private var dispatcher: Dispatcher!
+
+    /// Mock Storage: InMemory
+    ///
+    private var storageManager: MockStorageManager!
+
+    /// Mock Network: Allows us to inject predefined responses!
+    ///
+    private var network: MockNetwork!
+
+    /// Convenience Property: Returns the StorageType associated with the main thread.
+    ///
+    private var viewStorage: StorageType {
+        return storageManager.viewStorage
+    }
+
+    /// Testing SiteID
+    ///
+    private let sampleSiteID: Int64 = 12345
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockStorageManager()
+        network = MockNetwork()
+    }
+
+    func test_synchronizeSitePlugins_stores_plugins_correctly() {
+        // Given
+        network.simulateResponse(requestUrlSuffix: "plugins", filename: "plugins")
+        let store = SitePluginStore(dispatcher: dispatcher, storageManager: storageManager, network: network)
+
+        // When
+        let result: Result<Void, Error> = waitFor { promise in
+            let action = SitePluginAction.synchronizeSitePlugins(siteID: self.sampleSiteID) { result in
+                promise(result)
+            }
+            store.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertEqual(viewStorage.countObjects(ofType: StorageSitePlugin.self), 5) // number of plugins in json file
+    }
+
+    func test_synchronizeSitePlugins_removes_stale_plugins_correctly() {
+        // Given
+        let stalePlugin = SitePlugin.fake().copy(siteID: sampleSiteID, name: "Stale Plugin")
```

Comment:
How do you feel about these functions to create fake objects for tests?

Reply:
I like the convenience of these functions - I've been using [Fakery](https://github.com/vadymmarkov/Fakery) in my unit tests at work but I prefer the code generation solution we have here - there's no need for 3rd party libraries.

Reply:
Oh actually we're still using [Sourcery](https://github.com/krzysztofzablocki/Sourcery)... but with Sourcery Pro out now, we may use the extension to replace it.

Reply:
Good to know, I haven't explored in detail `Sourcery Pro` yet 😅